platform/security/sts/security-sts-realm/src/main/java/ddf/security/realm/sts/SamlRealm.java

```diff
@@ -0,0 +1,187 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.security.realm.sts;
+
+import ddf.security.assertion.SecurityAssertion;
+import ddf.security.assertion.saml.impl.SecurityAssertionSaml;
+import java.security.Principal;
+import java.util.List;
+import java.util.Optional;
+import org.apache.cxf.ws.security.tokenstore.SecurityToken;
+import org.apache.shiro.authc.AuthenticationException;
+import org.apache.shiro.authc.AuthenticationInfo;
+import org.apache.shiro.authc.AuthenticationToken;
+import org.apache.shiro.authc.SimpleAuthenticationInfo;
+import org.apache.shiro.authc.credential.CredentialsMatcher;
+import org.apache.shiro.realm.AuthenticatingRealm;
+import org.apache.shiro.subject.PrincipalCollection;
+import org.apache.shiro.subject.SimplePrincipalCollection;
+import org.codice.ddf.platform.filter.AuthenticationFailureException;
+import org.codice.ddf.security.handler.api.BaseAuthenticationToken;
+import org.codice.ddf.security.handler.api.SAMLAuthenticationToken;
+import org.codice.ddf.security.saml.assertion.validator.SamlAssertionValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SamlRealm extends AuthenticatingRealm {
+  private static final Logger LOGGER = (LoggerFactory.getLogger(SamlRealm.class));
+
+  private SamlAssertionValidator samlAssertionValidator;
+
+  private List<String> usernameAttributeList;
+
+  public SamlRealm() {
+    setCredentialsMatcher(new STSCredentialsMatcher());
+  }
+
+  /** Determine if the supplied token is supported by this realm. */
+  @Override
+  public boolean supports(AuthenticationToken token) {
+    boolean supported =
+        token != null && token.getCredentials() != null && token instanceof SAMLAuthenticationToken;
+
+    if (supported) {
+      LOGGER.debug("Token {} is supported by {}.", token.getClass(), SamlRealm.class.getName());
+    } else if (token != null) {
+      LOGGER.debug("Token {} is not supported by {}.", token.getClass(), SamlRealm.class.getName());
+    } else {
+      LOGGER.debug("The supplied authentication token is null. Sending back not supported.");
+    }
+
+    return supported;
+  }
+
+  /** Perform authentication based on the supplied token. */
+  @Override
+  protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken token) {
+    Object credential = null;
+
+    // perform validation
+    if (token instanceof SAMLAuthenticationToken) {
```

Comment:
❓ Shouldn't this test be unnecessary? The `supports` method would have been called first and this would only be invoked if that returned true, correct?